File: backend-play/app/com/github/brane08/pagila/store/entities/City.java
package com.github.brane08.pagila.store.entities;

import com.github.brane08.pagila.seedworks.entities.BaseModel;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "city")
public class City extends BaseModel {
	@Id
	@Column(name = "city_id")
	public Integer cityId;

	@Column(name = "city")
	public String city;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "country_id", columnDefinition = "int2")
	public Country country;

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || getClass() != o.getClass()) return false;
		City that = (City) o;
		return Objects.equals(city, that.city) && Objects.equals(country, that.country);
	}

	@Override
	public int hashCode() {
		return Objects.hash(city, country);
	}
}

File: backend-micronaut/src/main/kotlin/com/github/brane08/pagila/models/City.kt
package com.github.brane08.pagila.models

import javax.persistence.*

@Entity
@Table(name = "city")
class City : BaseModel() {
	@Id
	@Column(name = "city_id")
	var cityId: Int? = null

	@Column(name = "city")
	var city: String? = null

	@ManyToOne(optional = false)
	@JoinColumn(name = "country_id", columnDefinition = "int2")
	var country: Country? = null
}